app/submitter/converter.py

```diff
@@ -82,14 +83,30 @@ def convert_answers_to_data(answer_store, questionnaire_schema):
     for answer in answer_store.answers:
         item = questionnaire_schema.get_item_by_id(answer['answer_id'])
         value = answer['value']
+
         if item is not None and value is not None:
-            if item.code not in data:
-                data[item.code] = _encode_value(value)
+            if not isinstance(item, CheckboxAnswer) or any('q_code' not in option for option in item.options):
+
+                if item.code not in data:
```

Comment:
Unless I've misunderstood, there's no need to modify the original list, you just need to inspect it and return a variable based on it's contents. Something like:

    def _get_answer_data(original_data, item_code):
        if item_code in original_data:
            data_to_return = original_data[item_code]
            if not isinstance(data_to_return, list):
                 data_to_return = [data_to_return]
            data_to_return.append(_encode_value(value))

            return data_to_return
        return _encode_value(value)


    data[item.code] = _get_answer_data(data, item.code)